Tests/SentryTests/SentryScopeSwiftTests.swift

```diff
@@ -238,6 +238,31 @@ class SentryScopeSwiftTests: XCTestCase {
         XCTAssertEqual(context as? [String: [String: String]],
                        actual?.context as? [String: [String: String]])
     }
+    
+    func testApplyToEvent_EventWithContext_MergesContext() {
+        let context = NSMutableDictionary(dictionary: [
+            "first": ["a": "b", "c": "d"]])
+        let event = fixture.event
+        event.context = context as? [String: [String: String]]
+        
+        let expectedAppContext = [
+            "first": [ "a": "b", "c": "d", "e": "f"],
+            "second": ["0": "1" ]
+        ]
+        
+        let scope = fixture.scope
+        scope.setContext(value: ["e": "f"], key: "first")
+        scope.setContext(value: ["0": "1"], key: "second")
+        
+        let actual = scope.apply(to: event, maxBreadcrumb: 10)
+        let actualContext = actual?.context as? [String: [String: String]]
+        
+        context.addEntries(from: fixture.context)
+        context.addEntries(from: expectedAppContext)
+        
+        XCTAssertEqual(context as? [String: [String: String]],
+                       actualContext)
+    }
```

Comment:
m: if we decide to keep the current overwrite logic for unequal types, we should also write a test case for that here.

Reply:
Quick question since Im no specialist in tests. If the other test succeeds and keeps the behaviour at check, wouldn't be just a repetition to have the same test here? Here we just need to make sure both `context`  are being merged, or Am I missing something?

Reply:
I'm thinking that if someone comes in later and doesn't know that data can be dropped from the scope dictionaries this way, they might build on that assumption which could break other functionality in an unexpected way, either at the time they introduce their change, or when a change comes in later that also uses that behavior. It would be helpful to have a test that can fail in that specific case.

Reply:
Done